### PR TITLE
*: Hide "Mem" column from SHOW PROCESSSLIST

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve? 

#8482

### What is changed and how it works?

The command `SHOW PROCESSLIST` has a column called 'Mem', which at least for me has always showed a zero value.  This patch removes the column completely, but an alternative fix (beyond my current skill) could be to fix the output to be correct.

In MySQL the memory info is available in perfschema (try for example: `SELECT * FROM sys.session`) , but since TiDB does not implement perfschema maybe extending processlist is the best way to present it?  I could be convinced either way, but I am submitting this patch because a zero value is a bad first experience.

MySQL 5.7.40:
```
mysql57> mysql57> SHOW PROCESSLIST;
+----+----------+-----------+------+---------+------+----------+------------------+
| Id | User     | Host      | db   | Command | Time | State    | Info             |
+----+----------+-----------+------+---------+------+----------+------------------+
|  2 | msandbox | localhost | NULL | Query   |   -1 | starting | SHOW PROCESSLIST |
+----+----------+-----------+------+---------+------+----------+------------------+
1 row in set (0.00 sec)
```

TiDB master:
```
mysql> SHOW PROCESSLIST;
+------+------+-----------+------+---------+------+-------+------------------+------+
| Id   | User | Host      | db   | Command | Time | State | Info             | Mem  |
+------+------+-----------+------+---------+------+-------+------------------+------+
|    1 | root | 127.0.0.1 |      | Query   |    0 | 2     | SHOW PROCESSLIST |    0 |
+------+------+-----------+------+---------+------+-------+------------------+------+
1 row in set (0.00 sec)
```

TiDB master + this patch:
```
mysql> SHOW PROCESSLIST;
+------+------+-----------+------+---------+------+-------+------------------+
| Id   | User | Host      | db   | Command | Time | State | Info             |
+------+------+-----------+------+---------+------+-------+------------------+
|    2 | root | 127.0.0.1 |      | Query   |    0 | 2     | SHOW PROCESSLIST |
+------+------+-----------+------+---------+------+-------+------------------+
1 row in set (0.00 sec)
```

### Check List 

Tests 

 - Manual test

Ran manual test + fixed tests that broke.

Code changes

 - Has interface methods change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation
 - Need to be included in the release note

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8543)

### CODE REVIEW

**Comment by morgo** [comment]
@shenli I've updated the description to make it clearer.  'Mem' is not a column in MySQL - it was a TiDB extension not fully implemented.

File: session/session.go [hunk 1]
```diff
@@ -1504,7 +1504,6 @@ func (s *session) ShowProcess() util.ProcessInfo {
 	tmp := s.processInfo.Load()
 	if tmp != nil {
 		pi = tmp.(util.ProcessInfo)
-		pi.Mem = s.GetSessionVars().StmtCtx.MemTracker.BytesConsumed()
```
Comment (eurekaka) [inline session/session.go]: The `Mem` is not 0 if you run a memory consuming query simultaneously with `SHOW PROCESSLIST`. For example:
```
mysql> SHOW PROCESSLIST;
+------+------+-----------+------+---------+------+-------+-----------------------------------------------------------------+--------+
| Id   | User | Host      | db   | Command | Time | State | Info                                                            | Mem    |
+------+------+-----------+------+---------+------+-------+-----------------------------------------------------------------+--------+
|    1 | root | 127.0.0.1 | tmp  | Query   |    0 | 2     | SHOW PROCESSLIST                                                |      0 |
|    2 | root | 127.0.0.1 | tmp  | Query   |    2 | 2     | insert into tbl select t1.a, t1.b, t1.c from tbl t1 join tbl t2 | 245616 |
+------+------+-----------+------+---------+------+-------+-----------------------------------------------------------------+--------+
2 rows in set (0.02 sec)
```

Comment (zz-jason) [reply session/session.go]: For now, we only trace the memory usage in some memory intensive SQL operators, for example, HashJoin/IndexLookupJoin/IndexLookupReader, etc.

Comment (eurekaka) [reply session/session.go]: Oh, then I prefer removing this `Mem` field since it may confuse users without this knowledge.

**Comment by jackysp** [comment]
How to get the values of `Mem` if we remove it? @morgo

**Comment by morgo** [comment]
@jackysp When I had written the patch, I thought they were always zero.  It turns out there are cases of non-zero :(

How about adding an extension `SHOW PROCESSLIST MEMINFO`, where the column is shown?

**Comment by zz-jason** [comment]
> @jackysp When I had written the patch, I thought they were always zero. It turns out there are cases of non-zero :(
> 
> How about adding an extension `SHOW PROCESSLIST MEMINFO`, where the column is shown?

Glade to see this new feature. 😄

**Pull request merged**